agent/bench-scripts/pbench-linpack

```diff
@@ -201,25 +209,56 @@ for thread in ${threads//,/ }; do
 	mkdir -p "${iteration_dir}"
 	record_iteration ${count} ${thread} ${iteration}
 	echo "Starting iteration ${iteration}"
-
-	# Pre-processing: generate the linpack.input dirs (in /var/lib/pbench-agent/tmp)
+	# Pre-processing: generate the linpack.input files
 	preprocess-iteration "${iteration_dir}" "${thread}"
+	# Distribute the linpack.input files to remote clients if necessary
+	for client in ${clients//,/ }; do
+		if ! pbench-is-local "${client}"; then
+			ssh ${ssh_opts} "${client}" "mkdir -p '${iteration_dir}'"
+			scp ${scp_opts} "${iteration_dir}/"linpack* "${client}:${iteration_dir}"
+		fi
+	done
 
 	for sample in $(seq 1 ${nr_samples}); do
 		echo "test sample ${sample} of ${nr_samples}"
 		sample_dir="${iteration_dir}/sample${sample}"
-		# Create the output directory expected by the postprocess script
-		mkdir -p "${sample_dir}/clients/localhost/"
 
 		# Run the benchmark and start/stop perf analysis tools
 		pbench-start-tools --group=${tool_group} --dir="${sample_dir}"
 
-		( cd "${iteration_dir}" && ./linpack.sh )
+		for client in ${clients//,/ }; do
+			if pbench-is-local "${client}"; then
+				( cd "${iteration_dir}" && screen -dmS pbench-linpack ./linpack.sh )
+			else
+				ssh ${ssh_opts} ${client} "cd '${iteration_dir}' && screen -dmS pbench-linpack './linpack.sh'"
+			fi
+		done
+
+		# Wait for all of them to finish
+		echo "Waiting for all clients to finish pbench-linpack jobs"
+		for client in ${clients//,/ }; do
+			if pbench-is-local "${client}"; then
+				while ! [ -e "${iteration_dir}/linpack.finished" ]; do
+					sleep 1
+				done
+			else
+				ssh ${ssh_opts} ${client} "while ! [ -e '${iteration_dir}/linpack.finished' ]; do sleep 1; done"
+			fi
+			echo "Iteration finished on ${client}"
+		done
 
 		pbench-stop-tools --group=${tool_group} --dir="${sample_dir}"
 
 		# Move the results to the output directory expected by the postprocess script
-		mv "${iteration_dir}/"linpack.{out,meta} "${sample_dir}/clients/localhost"
+		for client in ${clients//,/ }; do
+			client_result_dir="${sample_dir}/clients/${client}"
+			mkdir -p "${client_result_dir}"
+			if pbench-is-local "${client}"; then
+				mv "${iteration_dir}/"linpack.{out,meta} "${client_result_dir}"
```

Comment:
Based on local unit testing (which I'll post updates for in a PR shortly) this needs to be `cp` since the `linpack.meta` file is only created once per iteration and shared for each sample.

Reply:
Both the fix and the unit tests provided in PR https://github.com/ldoktor/pbench/pull/6.

Reply:
I see, it is actually semi-created during the execution as well (only the `threads` entry) which is why I used `mv`. I'm wondering whether it wouldn't be safer to always create it in `linpack.sh` during the execution, what do you think?

Reply:
That PR also fixes the hung test that occurs in the CI.

Reply:
Once we merge https://github.com/ldoktor/pbench/pull/6 into this PR we should be able to merge this to the `main` branch.